test/connectors/responses/GetStateBenefitsResponseSpec.scala

```diff
@@ -0,0 +1,137 @@
+          Right(Some(AllStateBenefitsData(StateBenefitsData(), None)))
+        )
+      }
+
+      "status is OK and invalid jsValue" in {
+        val jsValue: JsValue = Json.parse(
+          """
+            |{}
+            |""".stripMargin)
+
+        val httpResponse: HttpResponse = HttpResponse.apply(OK, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Left(ApiError(INTERNAL_SERVER_ERROR, SingleErrorBody.parsingError))
+        )
+      }
+
+      "status is NOT_FOUND and any jsValue" in {
+        val jsValue: JsValue = Json.parse(
+          """
+            |{}
+            |""".stripMargin)
+
+        val httpResponse: HttpResponse = HttpResponse.apply(NOT_FOUND, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Right(None)
+        )
+      }
+
+      "status is INTERNAL_SERVER_ERROR and jsValue for error" in {
+        val jsValue: JsValue = Json.toJson(SingleErrorBody("some-code", "some-reason"))
+
+        val httpResponse: HttpResponse = HttpResponse.apply(INTERNAL_SERVER_ERROR, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Left(ApiError(INTERNAL_SERVER_ERROR, SingleErrorBody("some-code", "some-reason")))
+        )
+      }
+
+      "status is SERVICE_UNAVAILABLE and jsValue for error" in {
+        val jsValue: JsValue = Json.toJson(SingleErrorBody("some-code", "some-reason"))
+
+        val httpResponse: HttpResponse = HttpResponse.apply(SERVICE_UNAVAILABLE, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Left(ApiError(SERVICE_UNAVAILABLE, SingleErrorBody("some-code", "some-reason")))
+        )
+      }
+
+      "status is BAD_REQUEST and jsValue for error" in {
+        val jsValue: JsValue = Json.toJson(SingleErrorBody("some-code", "some-reason"))
+
+        val httpResponse: HttpResponse = HttpResponse.apply(BAD_REQUEST, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Left(ApiError(BAD_REQUEST, SingleErrorBody("some-code", "some-reason")))
+        )
+      }
+
+      "status is UNPROCESSABLE_ENTITY and jsValue for error" in {
+        val jsValue: JsValue = Json.toJson(SingleErrorBody("some-code", "some-reason"))
+
+        val httpResponse: HttpResponse = HttpResponse.apply(UNPROCESSABLE_ENTITY, jsValue, anyHeaders)
+
+        underTest.read(anyMethod, anyUrl, httpResponse) shouldBe GetStateBenefitsResponse(
+          httpResponse,
+          Left(ApiError(UNPROCESSABLE_ENTITY, SingleErrorBody("some-code", "some-reason")))
+        )
+      }
+
+      "status is ANYTHING_ELSE and jsValue for error" in {
```

Comment:
It comes to my mind `OTHER` instead of `ANYTHING_ELSE`, as it is shorter? I am happy with the way it is too really, just throwing the idea in case you like it.

Reply:
OTHER is a shorter. I will change it